python/dgl/dataloading/limited_edge.py

```diff
@@ -0,0 +1,125 @@
+"""Limited edge dataloader."""
+# pylint: disable=bad-super-call, no-value-for-parameter
+from typing import Generic
+import functools
+
+import numpy as np
+import dgl
+from dgl.dataloading.dataloader import _TensorizedDatasetIter
+
+
+class LimitedEdgeDataloader(dgl.dataloading.NodeDataLoader):
+    """Limited Edge Dataloader class.
+
+    For the normal dataloader, each batch have a same batch size, which may cause the workload
+    unbanlance problem when some batches contains high degree nodes. The limited edge dataloader
+    can support different batch sizes among different batches. Users can parse two configures:
+    max_node and max_edge, which means the maximum number of nodes in a batch and the maximum
+    number of edges in a batch. The limited edge dataloader can provide the largest batch which
+    follows the rule.
+    """
+    def __init__(self, g, nids, sampler, max_node, max_edge, prefix_sum_in_degrees=None, \
+        device='cpu', shuffle=False, use_uva=False, drop_last=False, num_workers=0):
+
+        dataset = LimitedEdgeDataset(max_node, max_edge, g, nids, prefix_sum_in_degrees)
+        super().__init__(g,
+                         dataset,
+                         sampler,
+                         device=device,
+                         use_uva=use_uva,
+                         shuffle=shuffle,
+                         drop_last=drop_last,
+                         use_prefetch_thread=False,
+                         num_workers=num_workers)
+
+    def modify_max_edge(self, max_edge):
```

Comment:
Is this function used anywhere? Same for `modify_max_node` and `reset_batch_node`.

Reply:
These three functions will be used in the auto tuner (modify the `max_node` and `max_edge` after each batch according to the memory usage).

Reply:
Is autotuner included in this PR?

Reply:
No, here only expose the functions that the autotuner use. I will try to add it in this PR.